sql/src/main/java/io/crate/analyze/Analyzer.java

```diff
@@ -119,20 +157,14 @@ public Analyzer(Schemas schemas,
         this.restoreSnapshotAnalyzer = new RestoreSnapshotAnalyzer(repositoryService, schemas);
         this.createFunctionAnalyzer = new CreateFunctionAnalyzer();
         this.dropFunctionAnalyzer = new DropFunctionAnalyzer();
-        this.userManager = userManagerProvider.get();
         this.privilegesAnalyzer = new PrivilegesAnalyzer();
     }
 
     public Analysis boundAnalyze(Statement statement, SessionContext sessionContext, ParameterContext parameterContext) {
         Analysis analysis = new Analysis(sessionContext, parameterContext, ParamTypeHints.EMPTY);
-        try {
-            AnalyzedStatement analyzedStatement = analyzedStatement(statement, analysis);
-            userManager.ensureAuthorized(analyzedStatement, sessionContext);
-            analysis.analyzedStatement(analyzedStatement);
-        } catch (Exception e) {
-            userManager.validateException(e, sessionContext);
-            throw e;
-        }
+        AnalyzedStatement analyzedStatement = analyzedStatement(statement, analysis);
+        sessionContext.ensureStatementAuthorized(analyzedStatement);
+        analysis.analyzedStatement(analyzedStatement);
```

Comment:
hm not sure if I get your point. `analysis.analyzedStatement(analyzedStatement);` is setting the statement on the analysis...